frontend/src/metabase/nav/components/search/RecentsList/RecentsList.tsx

```diff
@@ -0,0 +1,164 @@
+    if (url) {
+      dispatch(push(url));
+    }
+  };
+
+  const onContainerClick = (item: RecentItem) => {
+    if (onClick) {
+      onClick({
+        ...item.model_object,
+        model: item.model,
+        name: item.model_object.display_name ?? item.model_object.name,
+        id: item.model_id,
+      });
+    } else {
+      onChangeLocation(item);
+    }
+  };
+
+  return (
+    <Paper withBorder className={className}>
+      {isRecentsListLoading ? (
+        <SearchLoadingSpinner />
+      ) : (
+        <Stack spacing="md" px="sm" py="md">
+          <Title order={4} px="sm">{t`Recently viewed`}</Title>
+          <Stack spacing={0}>
+            {wrappedResults.map((item, index) => {
+              const isActive = isItemActive(item);
+              const model = item.model;
+              const name = getItemName(item);
+              const moderated_status = getModeratedStatus(item);
+              const result = item;
+              const isSelected = cursorIndex === index;
+              const ref: Ref<HTMLButtonElement> = getRef(item) ?? null;
+
+              return (
+                <SearchResultContainer
+                  ref={ref}
+                  key={getItemKey(item)}
+                  component="button"
+                  onClick={() => onContainerClick(item)}
+                  isActive={isActive}
+                  isSelected={isSelected}
+                  p="sm"
+                >
+                  <ItemIcon active={isActive} item={result} type={model} />
+                  <ResultNameSection justify="center" spacing="xs">
+                    <Group spacing="xs" align="center" noWrap>
+                      <ResultTitle order={4} truncate>
+                        {name}
+                      </ResultTitle>
+                      <ModerationIcon status={moderated_status} size={14} />
+                    </Group>
+                    <SearchResultLink>
+                      {getTranslatedEntityName(item.model)}
+                    </SearchResultLink>
+                  </ResultNameSection>
+                  {isItemLoading(item) && (
+                    <LoadingSection px="xs">
+                      <Loader />
+                    </LoadingSection>
+                  )}
+                </SearchResultContainer>
+              );
+            })}
+          </Stack>
+        </Stack>
+      )}
+    </Paper>
+  );
+};
+
+const getItemKey = ({ model, model_id }: RecentItem) => {
+  return `${model}:${model_id}`;
+};
+
+const getItemName = ({ model_object }: RecentItem) => {
+  return model_object.display_name || model_object.name;
+};
+
+const getModeratedStatus = ({ model_object }: RecentItem) => {
+  return model_object.moderated_status;
+};
+
+const isItemActive = ({ model, model_object }: RecentItem) => {
+  if (model !== "table") {
+    return true;
+  }
+  return isSyncCompleted(model_object);
+};
+
+const isItemLoading = ({ model, model_object }: RecentItem) => {
+  if (model !== "table") {
```

Comment:
In `SearchResult` we also check for `"database"`, i.e. `if (result.model !== "database" && result.model !== "table") {`.

Is the difference intentional?

Reply:
I didn't write these - but I know that databases aren't listed in recent items so that's probably the reason why. I think some cleanup could be due here but let's leave that for another PR